RPIPs/RPIP-47.md

```diff
@@ -26,34 +26,34 @@ Additionally, the pDAO must have the ability to effect change in ways that benef
 This RPIP is part of a set of proposals motivated by a desire to rework Rocket Pool's tokenomics to ensure the protocol’s continued value, development, and longevity. For more details, see the supporting documentation [here](../tokenomics-explainers/001-why-rework). 
 
 ## Specification
-### Megapool delegates
-- `Megapool delegate` contracts SHALL have an expiration block (ie, execution layer block)
-- When a new `megapool delegate` is released, the contract upgrade SHALL include:
-  - Setting the new `megapool delegate`'s expiration block to "no expiration"
-  - Setting the previous `megapool delegate`'s expiration block to occur `delegate_upgrade_buffer` after the upgrade
-- A node operator SHALL be able to upgrade their active `megapool delegate` to the latest `megapool delegate` at any time (including after the active `megapool delegate`'s expiration block)
-- After the active `megapool delegate`'s expiration block:
-  - The active `megapool delegate` SHALL NOT be usable by the node operator
-  - The node operator SHALL NOT be able to claim any rewards
-- The initial setting of `delegate_upgrade_buffer` SHALL be 4 months.
-
-### Megapool recovery delegate
-- There SHALL be a single `megapool recovery delegate` contract that is used by all megapools in the protocol
-- The `megapool recovery delegate` for a megapool SHALL NOT be usable by anyone if the current block is less than or equal to the active `megapool delegate`'s expiration block
-- The `megapool recovery delegate` for a megapool SHALL be usable by any address if the current block is greater than the active `megapool delegate`'s expiration block
-- The `megapool recovery delegate` SHALL have the following capabilities:
-  - Force exiting ALL validators under the megapool
-    - In the Saturn 1 release, this MAY be a non-functional stub
-  - Removal of any pending validators from the Rocket Pool queue
-  - Processing of voluntary exit proofs
-  - Collection of any penalties or deficits
-  - Final distribution of rewards and capital 
-- The `megapool recovery delegate` SHOULD be updated only rarely, and only to support changes to the Ethereum execution layer exit process, or major changes in the Rocket Pool protocol. Simultaneous upgrades to both the `megapool recovery delegate` and the latest `megapool delegate` SHOULD be avoided if possible.
+
+### Upgrade Buffer Enforcement
+- Each `Megapool delegate` contract SHALL have an execution layer `expiration_block` variable which should initialize to "no expiration"
+- Each `Megapool delegate` contract SHALL have a non-modifiable `upgrade_buffer` value.
+  - The `Megapool delegate`'s `upgrade_buffer` SHALL be 120 days.
+- The only permitted modification to a `Megapool delegate`'s `expiration_block` variable SHALL be:
+  - For the pDAO, oDAO, or Security Council to set its value to now + its `upgrade_buffer`
+  - For the pDAO, oDAO, or Security Council to set its value to "no expiration"
```

Comment:
Taking suggestions for who should be able to do these. Currently all just because it was simple.

Reply:
What's the scenario where we'd want to set it back to "no expiration"? Can we just make this "network contract" (see also below)? oDAO would have to vote on it, Security Council could veto, pDAO has authority over upgrades.

Reply:
Back to no expiration would be if the new delegate was discovered to be bugged, etc. In a way where it wasn't easy to fix before the previous one expired. Expirations may not always be 120 day in the future.

Reply:
I'll update both to network contract.